include/xtensor-io/xblosc.hpp

```diff
@@ -0,0 +1,193 @@
+        }
+    }  // namespace detail
+
+    /**
+     * Save xexpression to blosc format
+     *
+     * @param stream An output stream to which to dump the data
+     * @param e the xexpression
+     */
+    template <typename E>
+    inline void dump_blosc(std::ofstream& stream, const xexpression<E>& e, int clevel=5, int doshuffle=1)
+    {
+        detail::dump_blosc_stream(stream, e, clevel, doshuffle);
+    }
+
+    /**
+     * Save xexpression to blosc format
+     *
+     * @param filename The filename or path to dump the data
+     * @param e the xexpression
+     */
+    template <typename E>
+    inline void dump_blosc(const std::string& filename, const xexpression<E>& e, int clevel=5, int doshuffle=1)
+    {
+        std::ofstream stream(filename, std::ofstream::binary);
+        if (!stream.is_open())
+        {
+            std::runtime_error("IO Error: failed to open file");
+        }
+        detail::dump_blosc_stream(stream, e, clevel, doshuffle);
+    }
+
+    /**
+     * Save xexpression to blosc format in a string
+     *
+     * @param e the xexpression
+     */
+    template <typename E>
+    inline std::string dump_blosc(const xexpression<E>& e, int clevel=5, int doshuffle=1)
+    {
+        std::stringstream stream;
+        detail::dump_blosc_stream(stream, e, clevel, doshuffle);
+        return stream.str();
+    }
+
+    /**
+     * Loads a blosc file
+     *
+     * @param stream An input stream from which to load the file
+     * @tparam T select the type of the blosc file
+     * @tparam L select layout_type::column_major if you stored data in
+     *           Fortran format
+     * @return xarray with contents from blosc file
+     */
+    template <typename T, layout_type L = layout_type::dynamic>
+    inline auto load_blosc(std::istream& stream)
+    {
+        std::size_t uncompressed_size;
+        T* uncompressed_buffer = reinterpret_cast<T*>(detail::load_blosc_file(stream, uncompressed_size));
+        std::size_t size = uncompressed_size / sizeof(T);
+        std::vector<std::size_t> shape = {size};
+        auto array = adapt(uncompressed_buffer, size, no_ownership(), shape);
+        return array;
+    }
+    /**
+     * Loads a blosc file
+     *
+     * @param filename The filename or path to the file
+     * @tparam T select the type of the blosc file
+     * @tparam L select layout_type::column_major if you stored data in
+     *           Fortran format
+     * @return xarray with contents from blosc file
+     */
+    template <typename T, layout_type L = layout_type::dynamic>
+    inline auto load_blosc(const std::string& filename)
+    {
+        std::ifstream stream(filename, std::ifstream::binary);
+        if (!stream.is_open())
+        {
+            std::runtime_error("load_blosc: failed to open file " + filename);
+        }
+        return load_blosc<T, L>(stream);
+    }
+
+    struct blosc_config
+    {
+        int clevel;
+        int doshuffle;
+    };
+
+    class xblosc
```

Comment:
Formatting: the implementation should be outside of the class definition. Similarly, the previous methods should be first delcared, and implemented after everything has been declared. This would also allow to move the detail section after the public API declaration.